procedures/crates.md

```diff
@@ -0,0 +1,140 @@
+
+When considering creating an out-of-tree crate, it is worth balancing how general the crate should
+be with the increased maintenance burden that this may bring if widely used.
+
+### Where should compiler crates live?
+Out-of-tree compiler crates should be hosted in the `rust-lang` organization - this simplifies
+integration with external infrastructure tooling and will inherit existing team permissions on
+GitHub. It should be made clear in any documentation that the compiler team and any appropriate
+working groups are responsible for the crate. It is not recommended to start with a prototype in
+another organization or personal repository.
+
+### Can existing out-of-tree crates from personal accounts or other organizations be transferred?
+Yes, this is encouraged. In order to do this, discuss this with the team and familiarize yourself
+with [the GitHub documentation for repository transfers][repo_transfers] and then arrange to perform
+the transfer. Once a transfer is complete, a redirect will exist in the original account or
+organization and this will conflict with the names of any new forks of the transferred repository -
+an email to GitHub is required to resolve this.
+
+[repo_transfers]: https://help.github.com/en/articles/transferring-a-repository
+
+### Who owns these crates?
+It is desired that a compiler team member or working group has loose ownership over a crate so
+that there are clear owners who are responsible for making sure that new versions are published and
+that pull requests are reviewed.
+
+### What should these crates be named?
+Crate naming will be discussed when new out-of-tree crates are proposed to the compiler team.
+
+Crate naming will differ on a case-by-case basis. Crates that are inherently tied to the
+compiler would benefit from a name that is prefixed with `rustc_`. This is an indicator of how
+stable the crate may be to prospective users. Other crates, which are more general-purpose, will
+have names that are disentangled from the compiler.
+
+### Are there any limitations on the review policy for out-of-tree crates?
+Generally, the working groups and team members that are primarily free to maintain the crate using
+whatever practices are best suited to their group, however, there are some limitations so that there
+is some uniformity across the compiler and out-of-tree crates:
+
+- Every compiler team member must be able to review and approve PRs.
+- Where possible, only active participants in the crate (or related working group) need be on the
+  highfive rotation for the crate.
+- Working group members (or regular contributors to the crate) who cannot approve PRs on rustc can
+  be allowed approve PRs to the crate.
+- Major pull requests should have multiple reviewers.
+
+### What is required of an out-of-tree crate?
+It is required that out-of-tree crates must:
+
+- Be dual-licensed with Apache 2.0 and MIT (as the compiler is) unless otherwise
```

Comment:
Note, that tidy lint has a source-of-truth list of licenses that are acceptable for external libraries. Perhaps mandate A2.0/MIT only for the crates that are truly a part of compiler (e.g. `rustc_data_structures` as opposed to some generic graph library?)

Reply:
It looks like that list is almost entirely composed of [combinations of Apache 2.0 and MIT](https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/deps.rs#L10-L19) with the only exception being unlicense - which I guess I could mention here if we felt it was worth it?

Reply:
I’m somewhat confident that licenses such as ISC would get added there in short order, if it was necessary for a desirable crate. The list is updated on-demand, rather than being an exhaustive list of licenses that we will ever accept.

Reply:
That's a great point, I've updated the document. Thanks.